src/display/base_factory.js

```diff
@@ -45,9 +45,8 @@ class BaseCanvasFactory {
   }
 
   create(width, height) {
-    if (width <= 0 || height <= 0) {
-      throw new Error("Invalid canvas size");
-    }
+    width = Math.max(width, 0);
+    height = Math.max(height, 0);
```

Comment:
It seems quite unfortunate to modify a general utility such as `CanvasFactory` just to fix one annotation bug.

Could we perhaps instead return an *empty* `operatorList`, from the src/core/annotation.js code, when the annotation has "bad" dimensions?

Reply:
I have been considering that option too and went a bit back and forth on where the responsibility of this should be in the code. In the end I also went for the idea of changing the `CanvasFactory` because I didn't really see a reason to disallow e.g. creating 0 x 0 canvases (even though it's probably not very useful to have such canvases, but breaking on it also seemed a bit unnecessary) and it might fix a whole class of similar bugs if we fix this in the factory itself as it's used in various places. However, I can also see the benefit of solving this locally for annotations only because it makes the scope smaller and that's the specific issue we have seen here.

In short, I don't really have a clear preference for either solution and think both could be fine here.

Reply:
To me it makes no sense to create a `0 x 0` canvas, since that basically points to a bug elsewhere, and these changes also make the `create` and `reset` methods *inconsistent* w.r.t. dimension validation which seems quite unfortunate.

Given that we ignore empty annotations in the display layer, [see here](https://github.com/mozilla/pdf.js/blob/a208d6bca72f8cb700f7bb506ea1bc3e76cbaec0/src/display/annotation_layer.js#L2986-L2989), it seems reasonable to me to generate *empty* operatorLists for such cases as well.

Reply:
That's a good point. I didn't know we already ignored empty annotations in that code already, so in that case the empty operator list solution indeed seems overall more consistent here.